android/app/src/main/java/app/organicmaps/util/HttpClient.java

```diff
@@ -71,6 +75,11 @@ public static Params run(@NonNull final Params p) throws IOException, NullPointe
     {
       connection = (HttpURLConnection) new URL(p.url).openConnection();
 
+      // Fix missing root certificates for HTTPS connections on Android 7 and below:
+      // https://community.letsencrypt.org/t/letsencrypt-certificates-fails-on-android-phones-running-android-7-or-older/205686
+      if (android.os.Build.VERSION.SDK_INT <= android.os.Build.VERSION_CODES.N && p.url.startsWith("https://"))
+        ((HttpsURLConnection) connection).setSSLSocketFactory(MwmApplication.sslSocketFactory);
```

Comment:
nit: Factor out a utility function to avoid copy-paste here and above?

Reply:
Introducing a utility function will add more lines of code than these 4 lines. It's a simple if, it may be harder to read in a utility function.

Reply:
It is not about lines of code, but keeping not obvious hacks in one place with comments, instead of 2-3 ...

Reply:
After a second thought I agree that it can be done cleaner. Thanks for the hints! @vng PTAL